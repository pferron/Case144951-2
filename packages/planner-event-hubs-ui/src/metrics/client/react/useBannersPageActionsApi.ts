/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface AssignPagesButtonClicked extends TrackProperties {}
export interface BannerContentSectionChecked extends TrackProperties {
  // Title provided for the banner
  bannerTitle: string;
  // Description provided for the banner
  bannerBody?: string;
  // Button text if there are any buttons added to the banner
  buttonText?: string;
  // Destination page or link on click on the button
  whereWillThisButtonBringYourMembers?: string;
  // Name of the page that button takes when selected Events+ hub
  selectAPage?: string;
  // External Link to which the button clicks takes to
  externalLink?: string;
}
export interface BannerDesignSectionChecked extends TrackProperties {
  // Text alignment selected for the banner
  textAlignment: string;
  // Font color of the text that appears on the banner
  fontColor?: string;
}
export interface BannerImageChecked extends TrackProperties {}
export interface BannerNameProvided extends TrackProperties {
  // Template chosen for the banner
  bannerTemplate: string;
  // Name provided for the banner
  bannerName: string;
}
export interface CreateBannerButtonClicked extends TrackProperties {}
export interface PagesAssignButtonClicked extends TrackProperties {
  // Name of the page assigned to the banner
  bannerPlacement: string;
}

interface AnalyticsApi {
  assignPagesButtonClicked: (
    props?: AssignPagesButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerContentSectionChecked: (
    props: BannerContentSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerDesignSectionChecked: (
    props: BannerDesignSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerImageChecked: (props?: BannerImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
  bannerNameProvided: (props: BannerNameProvided, options?: Rudder.Options, callback?: ApiCallback) => void;
  createBannerButtonClicked: (
    props?: CreateBannerButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  pagesAssignButtonClicked: (props: PagesAssignButtonClicked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBannersPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBannersPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who has intent to assign pages to banners in order to show banners on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : assign pages button clicked
   * downstreamName: assign pages button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign pages' in page section of banner
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {AssignPagesButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const assignPagesButtonClicked = useCallback(
    (props?: AssignPagesButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'assign pages button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'assign pages button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provides banner details in order to update the banner content section
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner content section checked
   * downstreamName: banner content section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner content section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerContentSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerContentSectionChecked = useCallback(
    (props: BannerContentSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner content section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner content section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who updates design section in order to change the text alignments in banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner design section checked
   * downstreamName: banner design section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner design section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerDesignSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerDesignSectionChecked = useCallback(
    (props: BannerDesignSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner design section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner design section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use in banners
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner image checked
   * downstreamName: banner image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in banners section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerImageChecked = useCallback(
    (props?: BannerImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provide a name to banners in order to create the banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner name provided
   * downstreamName: banner name provided
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerNameProvided} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerNameProvided = useCallback(
    (props: BannerNameProvided, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner name provided',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner name provided'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who wants to create banner in order to use in eventsplus pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : create banner button clicked
   * downstreamName: create banner button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when the user clicks on Create Banner' button on the banners page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {CreateBannerButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const createBannerButtonClicked = useCallback(
    (props?: CreateBannerButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'create banner button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'create banner button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who assigns the pages to banners in order to show the banner on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : pages assign button clicked
   * downstreamName: pages assign button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign' in the select 'Pages' modal
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {PagesAssignButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const pagesAssignButtonClicked = useCallback(
    (props: PagesAssignButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'pages assign button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'pages assign button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    assignPagesButtonClicked,
    bannerContentSectionChecked,
    bannerDesignSectionChecked,
    bannerImageChecked,
    bannerNameProvided,
    createBannerButtonClicked,
    pagesAssignButtonClicked
  };
};
