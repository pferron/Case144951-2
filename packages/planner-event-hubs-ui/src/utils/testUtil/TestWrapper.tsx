import React, { useMemo } from 'react';
import { LocalizationProvider } from 'nucleus-text';
import { Breakpoints } from '@cvent/carina/components/Breakpoints';
import { AppFeature } from '@cvent/planner-event-hubs-model/types';
import { ThemeProvider } from '@cvent/carina/components/ThemeProvider/ThemeProvider';
import { getDefaultTheme } from '@cvent/carina/utils/tokens';
import { AppFeaturesProvider } from '@components/AppFeaturesProvider';
import { UserPermissionsContext } from '@hooks/UserPermissionsProvider';
import locales, { defaultLocale } from '../../../locales';

export function TestWrapper({ children, appFeatures }: Props): JSX.Element {
  const userPermissionsValue = useMemo(() => ({ VideoCenter: true }), []);
  return (
    <ThemeProvider theme={getDefaultTheme()}>
      <LocalizationProvider locale={defaultLocale} locales={locales}>
        <Breakpoints ssrSize="m" key={typeof window === 'undefined' ? 'ssr' : 'csr'}>
          <AppFeaturesProvider value={appFeatures}>
            <UserPermissionsContext.Provider value={userPermissionsValue}>{children}</UserPermissionsContext.Provider>
          </AppFeaturesProvider>
        </Breakpoints>
      </LocalizationProvider>
    </ThemeProvider>
  );
}

interface Props {
  children: JSX.Element;
  appFeatures?: AppFeature[];
}
