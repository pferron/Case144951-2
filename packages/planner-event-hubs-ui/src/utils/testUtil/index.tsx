import { render, configure } from '@testing-library/react';
import { TestWrapper } from './TestWrapper';

configure({ testIdAttribute: 'data-cvent-id' });

type RenderParams = Parameters<typeof render>;
type RenderReturn = ReturnType<typeof render>;
const customRender = (ui: RenderParams[0], options?: RenderParams[1]): RenderReturn =>
  render(ui, { wrapper: TestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
