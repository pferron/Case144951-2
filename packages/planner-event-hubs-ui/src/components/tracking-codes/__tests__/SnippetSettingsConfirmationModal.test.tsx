import React from 'react';
import { TestWrapper } from '@utils/testUtil/TestWrapper';
import { render, screen } from '@testing-library/react';
import SnippetSettingsConfirmationModal from '@components/tracking-codes/SnippetsSettingsConfirmatonModal';
import 'jest-axe/extend-expect';
import { axe } from 'jest-axe';

describe('SnippetSettingsConfirmationModal', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const mockFn = jest.fn();

  it('Renders SnippetSettingsConfirmation Modal', async () => {
    const { baseElement } = render(
      <TestWrapper>
        <SnippetSettingsConfirmationModal
          isNewSnippetAdded
          isOpen
          setIsModalOpen={mockFn}
          setSelectCodeSnippetModelOpen={mockFn}
          setSnippetSettingsModalIsOpen={mockFn}
        />
      </TestWrapper>
    );

    expect(baseElement).toMatchSnapshot();

    // Verify elements
    expect(await screen.findByRole('button', { name: 'Leave' })).toBeInTheDocument();
    expect(await screen.findByRole('button', { name: 'Stay' })).toBeInTheDocument();

    // Verify text
    expect(await screen.findByText('Leave without saving?')).toBeInTheDocument();
    expect(await screen.findByText('Leave')).toBeInTheDocument();
    expect(await screen.findByText('Stay')).toBeInTheDocument();
    expect(
      await screen.findByText('If you leave, your changes wonâ€™t be saved. Stay here to save them.')
    ).toBeInTheDocument();
    expect(await axe(baseElement)).toHaveNoViolations();
  });
});
