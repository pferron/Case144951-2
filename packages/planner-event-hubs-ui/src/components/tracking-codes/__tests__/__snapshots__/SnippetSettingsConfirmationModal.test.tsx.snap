// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnippetSettingsConfirmationModal Renders SnippetSettingsConfirmation Modal 1`] = `
.emotion-0 {
  position: fixed;
  height: 100%;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  top: 0;
  left: 0;
  z-index: 1100;
}

.emotion-1 {
  position: fixed;
  height: 100%;
  width: 100%;
  -webkit-backdrop-filter: blur(4px);
  backdrop-filter: blur(4px);
  background-color: rgba(217,220,222,0.8);
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  z-index: 1100;
  transition-property: opacity;
  transition-timing-function: cubic-bezier(0, 0, 0.4, 0.9);
  transition-duration: 200ms;
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2 {
  border: none;
  outline: none;
  box-sizing: border-box;
  max-height: 100vh;
  max-width: calc(100vw - 32px);
  background-color: rgb(255,255,255);
  overflow: hidden;
  box-shadow: 0px 4px 20px -2px rgba(0, 0, 0, 0.25);
  z-index: 1100;
  border-radius: 10px;
  width: 420px;
  transition-property: opacity,transform;
  transition-timing-function: cubic-bezier(0, 0, 0.4, 0.9);
  transition-duration: 200ms;
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  width: 25rem;
}

.emotion-3 {
  direction: ltr;
  box-sizing: border-box;
  height: 100%;
  width: 100%;
  overflow: hidden;
  transition-property: opacity;
  transition-timing-function: cubic-bezier(0, 0, 0.4, 0.9);
  transition-duration: 200ms;
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  transition-delay: 200ms;
}

.emotion-4 {
  margin: 1.5rem 1.5rem 1.5rem 1.5rem;
}

.emotion-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-height: inherit;
  max-width: inherit;
  height: 100%;
  width: 100%;
  overflow: hidden;
  box-sizing: border-box;
}

.emotion-6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  max-width: 100vw;
  max-height: 100vh;
  height: 100%;
  width: 100%;
  overflow-y: auto;
  position: relative;
}

.emotion-7 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  overflow: visible;
  -webkit-transition: box-shadow .2s ease-in-out;
  transition: box-shadow .2s ease-in-out;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 1rem;
  padding: 0.15rem;
}

.emotion-9 {
  font-size: 1.5rem;
  color: rgb(26,32,38);
  width: 100%;
  font-weight: 400;
}

.emotion-11 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding: 0.15rem;
}

.emotion-12 {
  direction: ltr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(0,106,225);
  border: 1px solid transparent;
  border-radius: 10px;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Rubik','Helvetica','Arial';
  font-style: normal;
  font-weight: 400;
  height: 40px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  letter-spacing: 0;
  margin-right: 0;
  margin-left: 0;
  max-width: 100%;
  padding-bottom: 0;
  padding-top: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color cubic-bezier(0, 0, 0.39, 1) 100ms,color cubic-bezier(0, 0, 0.39, 1) 100ms,border-color cubic-bezier(0, 0, 0.39, 1) 100ms,box-shadow cubic-bezier(0.2, 0, 0.4, 0.9) 100ms;
  transition: background-color cubic-bezier(0, 0, 0.39, 1) 100ms,color cubic-bezier(0, 0, 0.39, 1) 100ms,border-color cubic-bezier(0, 0, 0.39, 1) 100ms,box-shadow cubic-bezier(0.2, 0, 0.4, 0.9) 100ms;
  color: rgb(255,255,255);
  font-size: 1rem;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  min-width: 96px;
  margin-left: 0.5rem;
}

@media (hover: hover) {
  .emotion-12:hover {
    background-color: #004CD6;
  }
}

.emotion-12:active {
  background-color: #0025BD;
  -webkit-transform: scale(0.96);
  -moz-transform: scale(0.96);
  -ms-transform: scale(0.96);
  transform: scale(0.96);
}

.emotion-12:focus-visible {
  outline: none;
  box-shadow: 0 0 0 2px #4494EF;
}

.emotion-12 svg {
  fill: rgb(255,255,255);
}

.emotion-13 {
  min-width: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-14 {
  direction: ltr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 10px;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Rubik','Helvetica','Arial';
  font-style: normal;
  font-weight: 400;
  height: 40px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  letter-spacing: 0;
  margin-right: 0;
  margin-left: 0;
  max-width: 100%;
  padding-bottom: 0;
  padding-top: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color cubic-bezier(0, 0, 0.39, 1) 100ms,color cubic-bezier(0, 0, 0.39, 1) 100ms,border-color cubic-bezier(0, 0, 0.39, 1) 100ms,box-shadow cubic-bezier(0.2, 0, 0.4, 0.9) 100ms;
  transition: background-color cubic-bezier(0, 0, 0.39, 1) 100ms,color cubic-bezier(0, 0, 0.39, 1) 100ms,border-color cubic-bezier(0, 0, 0.39, 1) 100ms,box-shadow cubic-bezier(0.2, 0, 0.4, 0.9) 100ms;
  color: #006DE9;
  font-size: 1rem;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  min-width: 96px;
}

@media (hover: hover) {
  .emotion-14:hover {
    background-color: rgb(242,248,254);
    color: #004CD6;
  }

  .emotion-14:hover svg {
    fill: #004CD6;
  }
}

.emotion-14:active {
  background-color: #CFE4FC;
  color: #004CD6;
  -webkit-transform: scale(0.96);
  -moz-transform: scale(0.96);
  -ms-transform: scale(0.96);
  transform: scale(0.96);
}

.emotion-14:active svg {
  fill: #004CD6;
}

.emotion-14:focus-visible {
  outline: none;
  box-shadow: 0 0 0 2px #4494EF;
}

<body
  style="overflow: hidden;"
>
  <div />
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-cvent-id="snippet-settings-configure-confirmation-scrim"
    />
    <div
      aria-label="Leave without saving?"
      aria-modal="true"
      class="focus-visible emotion-2"
      data-cvent-id="snippet-settings-configure-confirmation"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <div
                class="emotion-7"
              >
                <div
                  class="emotion-8"
                >
                  <span
                    class="emotion-9"
                  >
                    Leave without saving?
                  </span>
                </div>
              </div>
              <div>
                If you leave, your changes wonâ€™t be saved. Stay here to save them.
              </div>
            </div>
            <div
              class="emotion-7"
            >
              <div>
                <div
                  class="emotion-11"
                >
                  <button
                    aria-label="Stay"
                    class="emotion-12"
                    data-cvent-id="snippet-settings-configure-confirmation-save-button"
                    type="button"
                  >
                    <span
                      class="emotion-13"
                    >
                      Stay
                    </span>
                  </button>
                  <button
                    aria-label="Leave"
                    class="emotion-14"
                    data-cvent-id="snippet-settings-configure-confirmation-cancel-button"
                    type="button"
                  >
                    <span
                      class="emotion-13"
                    >
                      Leave
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
